AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -105,27 +100,39 @@ - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
   return child;
 }
 
+#pragma mark - Parent
+
 - (void)setParent:(id<ASLayoutable>)parent
 {
-  // FIXME: Locking should be evaluated here.  _parent is not widely used yet, though.
-  _parent = parent;
+  {
+    ASDN::MutexLocker l(_propertyLock);
```

Comment:
We can use an instance variable with the name `_propertyLock` here as it's no subclass of `ASDisplayNode`. We probably should rename that too.